packages/workspace/test/workspace/path_index.test.ts

```diff
@@ -536,3 +536,227 @@ describe('updatePathIndex', () => {
+    const singleFieldObj = new ObjectType({
+      elemID: elemId,
+      fields: {
+        uriField: {
+          refType: createRefToElmWithValue(BuiltinTypes.STRING),
+          annotations: {
+            test: 'test',
+          },
+        },
+      },
+      path: ['salto', 'obj', 'field'],
+    })
+
+    const singleFieldObjAnnotations = new ObjectType({
+      elemID: elemId,
+      annotationRefsOrTypes: {
+        anno: createRefToElmWithValue(BuiltinTypes.STRING),
+      },
+      annotations: {
+        anno: 'Bye',
+      },
+      path: ['salto', 'obj', 'annotations'],
+    })
+    const pathHints = getElementsPathHints([singleFieldObj, singleFieldObjAnnotations])
+    expect(pathHints.length).toEqual(4)
+    expect(pathHints).toEqual(expect.arrayContaining([
+      expect.objectContaining({ key: 'salto.obj', value: [['salto', 'obj', 'field'], ['salto', 'obj', 'annotations']] }),
+      expect.objectContaining({ key: 'salto.obj.attr', value: [['salto', 'obj', 'annotations']] }),
+      expect.objectContaining({ key: 'salto.obj.annotation', value: [['salto', 'obj', 'annotations']] }),
+      expect.objectContaining({ key: 'salto.obj.field', value: [['salto', 'obj', 'field']] }),
+    ]))
+  })
+
+  it('should return path hints for divided annotations and divided fields in ObjectType', async () => {
+    const objFragStdFields = new ObjectType({
+      elemID: elemId,
+      fields: {
+        stdField: {
+          refType: createRefToElmWithValue(BuiltinTypes.STRING),
+          annotations: {
+            test: 'test',
+          },
+        },
+      },
+      path: ['salto', 'obj', 'standardFields'],
+    })
+    const objFragCustomFields = new ObjectType({
+      elemID: elemId,
+      fields: {
+        customField: {
+          refType: createRefToElmWithValue(BuiltinTypes.NUMBER),
+          annotations: {
+            test: 'test',
+          },
+        },
+      },
+      path: ['salto', 'obj', 'customFields'],
+    })
+    const objFragAnnotationsOne = new ObjectType({
+      elemID: elemId,
+      annotationRefsOrTypes: {
+        anno: createRefToElmWithValue(BuiltinTypes.STRING),
+      },
+      annotations: {
+        anno: 'Hey',
+      },
+      path: ['salto', 'obj', 'annotationsOne'],
+    })
+    const objFragAnnotationsTwo = new ObjectType({
+      elemID: elemId,
+      annotationRefsOrTypes: {
+        ping: createRefToElmWithValue(BuiltinTypes.STRING),
+      },
+      annotations: {
+        ping: 'pong',
+      },
+      path: ['salto', 'obj', 'annotationsTwo'],
+    })
+    const pathHints = getElementsPathHints([
+      objFragAnnotationsOne, objFragAnnotationsTwo, objFragStdFields, objFragCustomFields,
+    ])
+    expect(pathHints.length).toEqual(10)
+    expect(pathHints).toEqual(expect.arrayContaining([
+      expect.objectContaining({ key: 'salto.obj.annotation', value: [['salto', 'obj', 'annotationsOne'], ['salto', 'obj', 'annotationsTwo']] }),
```

Comment:
I think extracting these paths to variables would make the test more readable

e.g.
```suggestion
      expect.objectContaining({ key: 'salto.obj.annotation', value: [annotationsOnePath, annotationsTwoPath] }),
```